crates/throttle/src/throttle.rs

```diff
@@ -89,6 +151,38 @@ async fn redis_throttle(
     })
 }
 
+async fn gcra_throttle(
+    conn: RedisConnection,
+    key: &str,
+    limit: u64,
+    period: Duration,
+    max_burst: u64,
+    quantity: Option<u64>,
+) -> Result<ThrottleResult, Error> {
+    let mut script = GCRA_SCRIPT.prepare_invoke();
+    script
+        .key(key)
+        .arg(limit)
+        .arg(period.as_secs())
+        .arg(max_burst)
+        .arg(quantity.unwrap_or(1));
+
+    let result =  conn.invoke_script(script).await.context("error invoking redis GCRA script")?;
+    let result = <(u64, u64, u64, u64, String, String) as FromRedisValue>::from_redis_value(&result)?;
+
+    Ok(ThrottleResult {
+        throttled: result.0 == 1,
+        limit: result.1, // TODO: this value is probably wrong
```

Comment:
Need to fix this, wasn't yet obvious what to put in here but I can root around some more.